pkg/cmd/pulumi/packagecmd/package_add.go

```diff
@@ -174,8 +174,14 @@ from the parameters, as in:
 			}
 
 			// Build and add the package spec to the project
-			source := strings.Split(plugin, "@")[0]
-			version := pkg.Version.String()
+			pluginSplit := strings.Split(plugin, "@")
+			source := pluginSplit[0]
+			version := ""
+			if pkg.Version != nil {
+				version = pkg.Version.String()
+			} else if len(pluginSplit) == 2 {
```

Comment:
Can it ever be true that the version is passed in the CLI argument but is not in the schema? There won’t be a CLI version for local plugins I suppose.

Reply:
I think you could do e.g. `pulumi package add some-random-provider@0.10.0`, but `some-random-provider` does not return the version in the schema for whatever reason.  I think it would be free to do so, even though it might end up in warnings in the CLI.